src/terraboot/cluster.clj

```diff
@@ -480,16 +480,21 @@
                     :subnets private-subnets
                     :lifecycle {:create_before_destroy true}
                     :elb []
-                    :alb [{:name "internal-tasks"
-                           :internal true
-                           :listeners (map #(assoc % :account-number account-number) slave-alb-listeners)
-                           :subnets elb-private-subnets
-                           :security-groups (concat [(cluster-id-of "aws_security_group" "slave-alb-sg")]
-                                                    remote-default-sgs)}] })
-              (vpc/private-route53-record "slaves"
-                                          environment-dns
-                                          environment-dns-identifier
-                                          {:zone_id (remote-output-of "vpc" "private-dns-zone")
-                                           :alias {:name (cluster-output-of "aws_alb" "internal-tasks" "dns_name")
-                                                   :zone_id (cluster-output-of "aws_alb" "internal-tasks" "zone_id")
-                                                   :evaluate_target_health true}}))))))
+                    :alb (let [listeners (map #(assoc % :account-number account-number) slave-alb-listeners)]
+                           (if (seq listeners)
+                             [{:name "internal-tasks"
+                               :internal true
+                               :listeners listeners
+                               :subnets elb-private-subnets
+                               :security-groups (concat [(cluster-id-of "aws_security_group" "slave-alb-sg")]
+                                                        remote-default-sgs)}]
+                             []))})
+              (let [listeners (map #(assoc % :account-number account-number) slave-alb-listeners)]
+                (if (seq listeners)
```

Comment:
try (when-let [listeners (seq (map etc etc

Reply:
in fact, why is all that 'map' stuff there at all, listeners isn't used in the body:

(when (seq slave-alb-listeners)

do the same thing, right?